lhotse/cut.py

```diff
@@ -1365,6 +1369,82 @@ def filter_supervisions(
         )
         return new_cut
 
+    def merge_supervisions(self) -> "MonoCut":
+        """
+        Return a copy of the cut that has all of its supervisions merged into
+        a single segment.
+
+        The new start is the start of the earliest superivion, and the new duration
+        is a minimum spanning duration for all the supervisions.
+
+        The text fields are concatenated with a whitespace, and all other string fields
+        (including IDs) are prefixed with "cat#" and concatenated with a hash symbol "#".
+        This is also applied to ``custom`` fields. Fields with a ``None`` value are omitted.
+        """
+        # "m" stands for merged in variable names below
+
+        def merge(values: Iterable[str]) -> str:
+            # e.g.
+            # values = ["1125-76840-0001", "1125-53670-0003"]
+            # return "cat#1125-76840-0001#1125-53670-0003"
+            values = list(values)
+            if len(values) == 1:
+                return values[0]
+            return "#".join(chain(["cat"], values))
+
+        sups = self.supervisions
+
+        if len(sups) == 1:
+            return self
+
+        mstart = min(s.start for s in sups)
+        mend = max(s.end for s in sups)
+        mduration = add_durations(mend, -mstart, sampling_rate=self.sampling_rate)
+
+        custom_keys = set(
+            k for s in sups if s.custom is not None for k in s.custom.keys()
+        )
+        alignment_keys = set(
+            k for s in sups if s.alignment is not None for k in s.alignment.keys()
+        )
+
+        msup = SupervisionSegment(
+            id=merge(s.id for s in sups),
+            # For MonoCut, the recording_id is always the same.
+            recording_id=sups[0].recording_id,
+            start=mstart,
+            duration=mduration,
+            # For MonoCut, the channel is always the same.
+            channel=sups[0].channel,
+            text=" ".join(s.text for s in sups if s.text) or None,
+            speaker=merge(s.speaker for s in sups if s.speaker) or None,
```

Comment:
I don't understands how this becomes None if none of the speaker attributes are defined; it looks to me like it becomes "cat", because length=0 is not treated specially in merge()

Reply:
It is not super clean; the trick is that if merge() returns an empty string, it trigger `or None` which returns the right hand side arg (and the whole expression evaluates to `None`). I’ll modify it so that `merge` returns Optional[str] instead which should read better.

Reply:
My point was that I think when given the empty list, merge() may return "cat", not None.

Reply:
Good point, fixed.